fgrpc/grpcrunner_test.go

```diff
@@ -30,34 +30,133 @@ import (
 	"google.golang.org/grpc/health/grpc_health_v1"
 )
 
+var (
+	// Generated from "make cert"
+	caCrt  = "../cert-tmp/ca.crt"
+	svrCrt = "../cert-tmp/server.crt"
+	svrKey = "../cert-tmp/server.key"
+	// used for failure test cases
+	failCrt = "../missing/cert.crt"
+	failKey = "../missing/cert.key"
+)
+
 func TestGRPCRunner(t *testing.T) {
 	log.SetLogLevel(log.Info)
-	port := PingServer("0", "bar", 0)
-	destination := fmt.Sprintf("localhost:%d", port)
+	iPort := PingServer("0", "", "", "bar", 0)
+	iDest := fmt.Sprintf("localhost:%d", iPort)
+	sPort := PingServer("0", svrCrt, svrKey, "bar", 0)
+	sDest := fmt.Sprintf("localhost:%d", sPort)
 
-	opts := GRPCRunnerOptions{
+	ro := GRPCRunnerOptions{
 		RunnerOptions: periodic.RunnerOptions{
 			QPS:        100,
 			Resolution: 0.00001,
 		},
-		Destination: destination,
-		Profiler:    "test.profile",
+		Profiler: "test.profile",
 	}
-	res, err := RunGRPCTest(&opts)
-	if err != nil {
-		t.Error(err)
-		return
+
+	tests := []struct {
+		name       string
+		runnerOpts GRPCRunnerOptions
+		expect     bool
+	}{
+		{
+			name:       "valid insecure runner with payload",
+			runnerOpts: ro,
+			expect:     true,
+		},
+		{
+			name:       "valid secure runner",
+			runnerOpts: ro,
+			expect:     true,
+		},
+		{
+			name:       "invalid insecure runner to secure server",
+			runnerOpts: ro,
+			expect:     false,
+		},
+		{
+			name:       "valid secure runner using nil credentials to Internet https server",
+			runnerOpts: ro,
+			expect:     true,
+		},
+		{
+			name:       "invalid secure runner to insecure server",
+			runnerOpts: ro,
+			expect:     false,
+		},
+		{
+			name:       "invalid secure runner using test cert to https prefix Internet server",
+			runnerOpts: ro,
+			expect:     false,
+		},
+		{
+			name:       "invalid secure runner using test cert to no prefix Internet server",
+			runnerOpts: ro,
+			expect:     false,
+		},
+		{
+			name:       "invalid name in secure runner cert",
+			runnerOpts: ro,
+			expect:     false,
+		},
+		{
+			name:       "invalid cert for secure runner",
+			runnerOpts: ro,
+			expect:     false,
+		},
 	}
-	totalReq := res.DurationHistogram.Count
-	ok := res.RetCodes[grpc_health_v1.HealthCheckResponse_SERVING]
-	if totalReq != ok {
-		t.Errorf("Mismatch between requests %d and ok %v", totalReq, res.RetCodes)
+	for _, test := range tests {
+		switch {
```

Comment:
I didn't mean to have to put a big switch here :( why can't those be above directly ?